src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -70,9 +73,9 @@ protected ResolveOperationBase(ISharingLifetimeScope mostNestedLifetimeScope)
         public IEnumerable<ResolveRequestContextBase> InProgressRequests => RequestStack;
 
         /// <summary>
-        /// Gets the <see cref="System.Diagnostics.DiagnosticSource"/> for the operation.
+        /// Gets the <see cref="System.Diagnostics.DiagnosticListener"/> for the operation.
         /// </summary>
-        public DiagnosticSource DiagnosticSource { get; }
+        public DiagnosticListener DiagnosticSource { get; }
```

Comment:
`DiagnosticListener` is what exposes the `IsEnabled()` fast null check; `DiagnosticSource` (the base class) only exposes `IsEnabled(string)` without the faster check. I figured since this was internal, passing around the strong type rather than casting each time made sense.

Reply:
👍